src/pyfme/models/euler_flat_earth.py

```diff
@@ -16,315 +16,172 @@
-        (dx_dt, dy_dt, dz_dt) in (m/s, m/s, m/s).
-
-    See Also
-    --------
-    kaeq, lamceq
-
-    References
-    ----------
-    .. [1] B. Etkin, "Dynamics of Atmospheric Flight", Courier Corporation,
-        p. 149 (5.8 The Flat-Earth Approximation), 2012.
-
-    .. [2] M. A. Gómez Tierno y M. Pérez Cortés, "Mecánica del Vuelo", Garceta
-        Grupo Editorial, pp.18-25 (Tema 2: Ecuaciones Generales del Movimiento),
-        2012.
-    """
-
-    u, v, w = lin_vel
-    theta, phi, psi = euler_angles
-
-    dx_dt = np.cos(theta) * np.cos(psi) * u + (np.sin(phi) * np.sin(theta) *
-                                               np.cos(psi) - np.cos(
-        phi) * np.sin(psi)) * v + (np.cos(phi) *
-                                   np.sin(theta) * np.cos(psi) + np.sin(
-        phi) * np.sin(psi)) * w
-
-    dy_dt = np.cos(theta) * np.sin(psi) * u + (np.sin(phi) * np.sin(theta) *
-                                               np.sin(psi) + np.cos(
-        phi) * np.cos(psi)) * v + (np.cos(phi) *
-                                   np.sin(theta) * np.sin(psi) - np.sin(
-        phi) * np.cos(psi)) * w
-
-    dz_dt = - np.sin(theta) * u + np.sin(phi) * np.cos(theta) * v + \
-            np.cos(phi) * np.cos(theta) * w
-
-    return np.array([dx_dt, dy_dt, dz_dt])
+from pyfme.models.systems import DynamicSystem
+
+
+class EulerFlatEarth(DynamicSystem):
+
+    n_states = 12
+
+    def __init__(self, use_jacobian=None, integrator=None,
+                 **integrator_params):
+
+        # TODO: use jacobian when it is calculated
+        super().__init__(n_states=self.n_states, use_jacobian=use_jacobian,
+                         integrator=integrator)
+
+    def dynamic_system_state_to_full_system_state(self, mass, inertia, forces,
+                                                  moments):
+        full_system_state = {}
+
+        t = self.time
+        y = self.state
+        y_dot = self._dynamic_system_equations(t, y, mass, inertia, forces,
+                                              moments)
+
+        # TODO: define the rest of conversions
+        full_system_state['geodetic_coordinates'] = np.array([0., 0., y[11]])  # implement
+        full_system_state['geocentric_coordinates'] = np.zeros(3)  # implement
+        full_system_state['earth_coordinates'] = y[9:12]
+        full_system_state['euler_angles'] = y[6:9]
+        full_system_state['quaternions'] = np.zeros(4)  # implement
+        full_system_state['vel_body'] = y[0:3]
+        full_system_state['vel_NED'] = y_dot[9:12]
+        full_system_state['vel_ang'] = y[3:6]
+        full_system_state['accel_body'] = y_dot[0:3]
+        full_system_state['accel_NED'] = np.zeros(3)  # implement
+        full_system_state['accel_ang'] = y_dot[6:9]
+
+        return full_system_state
+
+    def full_system_state_to_dynamic_system_state(self, full_system_state):
+
+        state = np.zeros(self.n_states)
+
+        state[0:3] = full_system_state.__getattribute__('vel_body')
```

Comment:
If `full_system_state` is a `dict`, then why `.__getattribute__('X')` instead of `['X']`?